File: app/ros_sims/sortie_sim.jl
#!/usr/bin/env julia
# should be set and run as an executable

# this file is just used for testing ROSInterface.jl

using Pkg
Pkg.activate(Base.source_dir() * "/..")

using RobotOS
@rosimport std_msgs.msg: Bool
@rosimport geometry_msgs.msg: PoseStamped
rostypegen(@__MODULE__)
using .std_msgs.msg
using .geometry_msgs.msg

function callback(pose::PoseStamped, pub::Publisher{BoolMsg})
    println("Pose received!")
    println(pose)
    rossleep(1)
    publish(pub, BoolMsg(true)) # sortie finished
end

function main()
    init_node("sortie_sim")
    pub = Publisher{BoolMsg}("sortie_finished", queue_size=1)
    sub = Subscriber{PoseStamped}("latest_sample", callback, (pub,), queue_size=1)
    spin()
end

if !isinteractive()
    main()
end

File: app/ros/sortie_sim.py
#!/usr/bin/env python
# should be set and run as an executable

# this file is just used for testing ROSInterface.jl

import rospy
from std_msgs.msg import Bool
from geometry_msgs.msg import PoseStamped

def callback(pose, pub):
    rospy.sleep(1)
    pub.publish(True) # sortie finished

def main():
    rospy.init_node("sortie_sim")
    pub = rospy.Publisher("sortie_finished", Bool, queue_size=1)
    rospy.Subscriber("latest_sample", PoseStamped, callback, pub, queue_size=1)
    rospy.spin()

if __name__ == '__main__':
    main()
